vcd/datasource_vcd_nsxt_edgegateway_dns.go

```diff
@@ -0,0 +1,64 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdNsxtEdgeGatewayDns() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdNsxtEdgeGatewayDnsRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge gateway ID for DNS configuration",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Status of the DNS Forwarder. Defaults to `true`",
+			},
+			"listener_ip": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IP on which the DNS forwarder listens.",
+			},
+			"snat_rule_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "The value is `true` if a SNAT rule exists for the DNS forwarder.",
+			},
+			"snat_rule_ip_address": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The external IP address of the SNAT rule. (VCD 10.5.0+)",
+			},
+			"default_forwarder_zone": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The default forwarder zone.",
+				Elem:        defaultForwarderZone,
```

Comment:
These two `defaultForwarderZone` and `conditionalForwarderZone` have required attributes, should they only be pure Computed ones when they're used in the data source?

Reply:
Yep, added new variables for Datasource, thanks for pointing out.